var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -22,18 +23,61 @@ class Timemory(CMakePackage):
             submodules=True)
 
+    linux = False if platform == 'darwin' else True
+
+    variant('shared', default=True, description='Build shared libraries')
+    variant('static', default=False, description='Build static libraries')
     variant('python', default=True, description='Enable Python support')
-    variant('mpi', default=False, description='Enable MPI support')
-    variant('tau', default=True, description='Enable TAU support')
-    variant('papi', default=True, description='Enable PAPI support')
-    variant('cuda', default=True, description='Enable CUDA support')
-    variant('cupti', default=True, description='Enable CUPTI support')
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('tau', default=False, description='Enable TAU support')
+    variant('papi', default=linux, description='Enable PAPI support')
+    variant('ompt', default=True, description='Enable OpenMP tools support')
+    variant('cuda', default=linux, description='Enable CUDA support')
+    variant('cupti', default=linux, description='Enable CUPTI support')
+    variant('tools', default=True, description='Build/install command line tools')
+    variant('vtune', default=False, description='Enable VTune support')
     variant('upcxx', default=False, description='Enable UPC++ support')
-    variant('gotcha', default=True, description='Enable GOTCHA support')
-    variant('likwid', default=True, description='Enable LIKWID support')
+    variant('gotcha', default=linux, description='Enable GOTCHA support')
+    variant('likwid', default=linux, description='Enable LIKWID support')
     variant('caliper', default=True, description='Enable Caliper support')
+    variant('dyninst', default=linux,
+            description='Build dynamic instrumentation tools')
+    variant('examples', default=False, description='Build/install examples')
     variant('gperftools', default=True,
             description='Enable gperftools support')
+    variant('kokkos_tools', default=True,
+            description='Build generic kokkos-tools libraries, e.g. kp_timemory, kp_timemory_filter')
+    variant('kokkos_build_config', default=False,
+            description='Build pre-configured (i.e. dedicated) kokkos-tools libraries, e.g. kp_timemory_cpu_flops')
+    variant('build_caliper', default=True,
+            description='Build Caliper submodule instead of spack installation')
+    variant('build_gotcha', default=False,
+            description='Build GOTCHA submodule instead of spack installation')
+    variant('build_ompt', default=True,
+            description='Build OpenMP/OMPT submodule instead of spack installation')
+    variant('cuda_arch', default='auto', description='CUDA architecture name',
+            values=('auto', 'kepler', 'tesla', 'maxwell', 'pascal',
+                    'volta', 'turing'), multi=False)
+    variant('cpu_target', default='auto',
+            description='Build for specific cpu architecture (specify cpu-model)')
```

Comment:
Spack already has microarchitecture detection. I would use that instead of a variant. @alalazo or @becker33 can explain how it works.

Reply:
While that sounds interesting, I omitted listing the 20+ cpu variants handled here: https://github.com/NERSC/timemory/blob/develop/cmake/Modules/FindCpuArch.cmake
and the cuda values are specific to what is here: https://github.com/NERSC/timemory/blob/5c0f89880db25e31a0778d8252b05c4ad5e50803/cmake/Modules/CUDAConfig.cmake#L30.